datascience/tables.py

```diff
@@ -406,6 +413,12 @@ def relabel(self, column_label, new_label):
             self._formats[new_label] = formatter
         return self
 
+    def with_relabeling(self, column_label, new_label):
+        """Returns a new table with column_label changed to new_label."""
```

Comment:
This docstring should be expanded along the lines of the others.  It should describe that the parameters expected can be a single string, or a tuple of strings.  Perhaps it can also take a dict?  Alternatives to the name ``with_relabeling`` could be ``table.relabel``, ``table.rename`` or ``table.rename_columns``.